### DISCUSSION TITLE
Deploying Nuxt 3 application to Kubernetes without any downtime

### DISCUSSION THREAD

**Discussion by pkalisz** [nuxt/nuxt#30261] [Questions]
Hello there. Recently we have migrated from Nuxt 2 to Nuxt 3 with our application. As a matter of fact only partially, so part of the entire solution works on Nuxt 2 but most of the client facing stuff is already on Nuxt 3. Important thing is that both applications use SSR (or we think so, we don't force anything to use ssr, we're going with default settings). Thanks to such setup we have a chance to compare how both apps behave in prod environment. Either we are doing something wrong, or don't understand something, but it looks to us that Nuxt 3 application is way more problematic than Nuxt 2. 

Our entire solution runs on Kubernetes cluster, managed in AWS. So far out default deployment strategy was a rolling update, which for Nuxt 2 and other applications (written mostly using PHP) provided zero downtime experience. With Nuxt 3 it is no longer the case. The main difference we have noticed is that when we enter the Nuxt 2 app it downloads 5-6 js and css files, while the Nuxt 3 app downloads 50-60 js and css each at least. This is problematic when applications is updating on the cluster, becase there is a transition time when old version and new version is available and sometimes when user enters the new version and assets are downloaded after page has been loaded requests hit the old version ending with 404.

I'm wondering if anyone ever experienced a similar problem? I understand that maybe we will have change our deployment strategy, but maybe there is a way to package all those files into smaller number of bigger chunks? Or again, maybe we don't understand how Nuxt 3 is "doing things". Any advice would be appreciated.

**Comment by pkalisz** [comment]
Seriously? No one is hosting Nuxt3 application using Kubernetes cluster? No one has ever encountered such problem?

**Comment by yassilah** [comment]
Hi! I'm actually also deploying Nuxt that way and unless I'm misunderstanding your issue, it seems a simple way around that would be to build your Nuxt app in a temp folder and only replace the old version once the build is complete? If so, I was thinking of doing something simple like: https://stackblitz.com/edit/github-dwznsewd?file=nuxt.config.ts. Is that helpful at all for you?

**Comment by sewalsh** [comment]
2 steps to ensure no downtime:
- Deploy the `.output\public` folder to a CDN/different hostname https://nuxt.com/docs/api/nuxt-config#cdnurl
- Deploy the Nuxt 3 apps to the cluster

Don't clear out the legacy files on the CDN as you'll want these for clients that have an old version of the app loaded in their browser.
